plugins/module_utils/nbdkit/nbdkit.go

```diff
@@ -134,7 +137,7 @@ func (c *NbdkitConfig) RunNbdKitURI(diskName string) (*NbdkitServer, error) {
 	logger.Log.Infof("Command: %v", cmd)
 
 	time.Sleep(100 * time.Millisecond)
-	err = WaitForNbdkitURI("localhost", "10809", 30*time.Second)
+	err = WaitForNbdkitURI("localhost", c.Port, c.Timeout, c.RetryDelay)
```

Comment:
If you make the port as a variable, you have to set for nbdkit as well otherwise it's just waiting for the wrong port.